ui/packages/catalog-ui-search/src/main/webapp/js/model/QueryResponse.js

```diff
@@ -175,7 +176,7 @@ module.exports = Backbone.AssociatedModel.extend({
         metacardDefinitions.addMetacardDefinitions(resp.types);
         if (resp.results) {
             var queryId = this.getQueryId();
-            var selectedResultTemplate = this.get('selectedResultTemplate');
+            var selectedResultTemplate = ResultForm.getResultTemplatesProperties().filter(form => form.id === this.get('selectedResultTemplate'))[0];
```

Comment:
is `ResultForm.getResultTemplatesProperties().filter(form => form.id === this.get('selectedResultTemplate'))[0];` safe if it ends up being empty?

Reply:
The list non-filtered list should always have at least the one `All Fields` entry in it, and input values to that field are restricted to an Enum, so it shouldn't be possible, but I'll add a check there just in case a network issue eventually throws that assumption out the window or something

Reply:
Actually, even if that happens, it won't throw any errors at runtime
![image](https://user-images.githubusercontent.com/8922497/40743795-57a9050e-6407-11e8-8a3f-92e190e3d2bc.png)

And then `selectedResultTemplate` will be undefined which will be caught [here](https://github.com/codice/ddf/blob/master/ui/packages/catalog-ui-search/src/main/webapp/js/model/QueryResponse.js#L51) since that's the only thing `selectedResultTemplate` is used for